src/lib.rs

```diff
@@ -240,6 +248,52 @@ impl Opts {
   }
 }
 
+#[derive(Default)]
+pub struct WarningOpts {
+  pub all: bool,
+  pub match_only_vars: bool,
+  pub unused_defs: bool,
+}
+
+impl WarningOpts {
+  pub fn all() -> Self {
+    Self { all: true, match_only_vars: true, unused_defs: true }
+  }
+
+  // TODO(refactor): this kind of code does not look good
+  pub fn from_vec(&mut self, values: Vec<String>) -> Result<(), String> {
+    for value in values {
+      match value.as_ref() {
+        "all" => self.all = true,
+        "unused-defs" => self.unused_defs = true,
+        "match-only-vars" => self.match_only_vars = true,
+        other => return Err(format!("Unknown option '{other}'.")),
+      }
+    }
+    Ok(())
+  }
+
+  /// Removes warnings based on the enabled flags.
+  pub fn filter(&self, wrns: Vec<Warning>) -> Vec<Warning> {
```

Comment:
I think it would be better to have this be: `pub fn is_warning_enabled(&self, warn: &Warning) -> bool`.
Then you can do `wrns.iter().filter(|w| opt.is_warning_enabled(w))`.